lib/weaviate/schema.rb

```diff
@@ -45,7 +45,7 @@ def create(
         req.body["properties"] = properties unless properties.nil?
         if multi_tenant.is_a?(Hash)
           req.body["multiTenancyConfig"] = multi_tenant
-        elsif multi_tenant.present?
+        elsif multi_tenant && !(multi_tenant.respond_to?(:empty?) && multi_tenant.empty?)
```

Comment:
If you look at the method signature: `multi_tenant: nil`, so it's either `nil` or "whatever the user passes in" (which should be `true`.

Reply:
I'm wondering if we should just change the method signature to:
```
multi_tenant: false
```
and then in the method itself, always set the `multiTenancyConfig` to:
```
req.body["multiTenancyConfig"] = {enabled: multi_tenant}
```

Reply:
It seems like this code is allowing a Hash to be passed in and set instead of `{ enabled: true }`

```ruby
if multi_tenant.is_a?(Hash)
  req.body["multiTenancyConfig"] = multi_tenant
```

I think because this is valid, according to the docs:

```python
    multi_tenancy_config=Configure.multi_tenancy(
        enabled=True,
        auto_tenant_creation=True
    )
```

so looks like there are two options here we need to handle

Reply:
wdyt about:

```ruby
def create(
  multi_tenant: false,
  auto_tenant_creation: false,

# ...

req.body["multiTenancyConfig"] = { enabled: multi_tenant }
req.body["multiTenancyConfig"].merge!(auto_tenant_creation: auto_tenant_creation) if auto_tenant_creation

```

Reply:
@mculp I don't see the auto_tenant_creation option. I'm not sure what the Python client implement but this gem just interfaces with the REST API.

https://weaviate.io/developers/weaviate/config-refs/schema

Reply:
@andreibondarev I think it was added in 1.25. Check the curl example here:

https://weaviate.io/developers/weaviate/manage-data/multi-tenancy#automatically-add-new-tenants

Reply:
Yeah, if you'd like to change it -- let's do it!

Reply:
I checked that this did automatically create tenants when set up properly.

I left the `is_a?(Hash)` check in there for backwards compatibility. It seems like wheover added this is using the code like:

```ruby
schema.create({
  class_name: "Question",
  multi_tenant: { enabled: true, autoTenantCreation: true }
})
```

I added a shortcut:

```ruby
schema.create({
  class_name: "Question",
  multi_tenant: true,
  auto_tenant_creation: true
})
```

These result in the same `multiTenancyConfig`, and I added a spec for the new case.

It seems a little less than optimal, but I didn't want scope to creep too far. I'm mainly here about the `.present?`

That said, while I was testing, I found one more instance of `.blank?` (another Rails method, the inverse of `.present?`) and just changed it to `.nil?` (Ruby method).

While I was writing this code, I had a lot of ideas for refactoring/simplification, but wanted to go ahead and ship this before moving on to anything else.